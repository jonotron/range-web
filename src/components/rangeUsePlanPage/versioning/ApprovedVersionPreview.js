import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'

const Grid = styled.div`
  display: grid;
  grid-template-columns: 24px 96px 1fr;
`

export default function ApprovedVersionPreview({ plan }) {
  return (
    <Grid>
      <div>{plan.version}</div>
      <div>{plan.date}</div>
      <div>{plan.author}</div>
    </Grid>
  )
}

ApprovedVersionPreview.propTypes = {
  plan: PropTypes.shape({
    version: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired, // TODO: turn into an actual date
    author: PropTypes.string.isRequired
  }).isRequired
}
