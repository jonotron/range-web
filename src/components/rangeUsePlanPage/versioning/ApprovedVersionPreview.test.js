import React from 'react'
import { render } from '@testing-library/react'
import { defaultStory } from './ApprovedVersionPreview.story'

it('should display version, date, and author', () => {
  const { container, getByText } = render(defaultStory())

  expect(getByText('v2')).toBeTruthy()
  expect(getByText('June 30, 2019')).toBeTruthy()
  expect(getByText('Jane Doe')).toBeTruthy()

  expect(container.firstChild).toMatchSnapshot()
})
