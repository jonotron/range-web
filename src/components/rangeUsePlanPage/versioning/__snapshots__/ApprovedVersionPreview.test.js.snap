// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display version, date, and author 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}

<div
  class="c0"
>
  <div>
    v2
  </div>
  <div>
    June 30, 2019
  </div>
  <div>
    Jane Doe
  </div>
</div>
`;
